### PR TITLE
Commit Boost Acceptance Test for List Pub Keys

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
Add commit boost acceptance test for list pub keys API. Also fix the public key identifiers used by commit boost by introducing K256ArtifactSigner.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

## Testing

- [x] I thought about testing these changes in a realistic/non-local environment.

### CODE REVIEW

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/SignerForIdentifier.java [hunk 1]
```diff
@@ -46,15 +46,13 @@ public Optional<String> sign(final String identifier, final Bytes data) {
   /**
    * Sign data for given identifier and return ArtifactSignature. Useful for SECP signing usages.
    *
-   * @param <T> The specific type of ArtifactSignature
    * @param identifier The identifier for which to sign data.
    * @param data Bytes which is signed
    * @return Optional ArtifactSignature of type T. Empty if no signer available for given identifier
    */
-  @SuppressWarnings("unchecked")
-  public <T extends ArtifactSignature<?>> Optional<T> signAndGetArtifactSignature(
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/SignerForIdentifier.java:55]: Removing this means we need to cast in some places

Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/SignerForIdentifier.java:55]: This is currently used only at one place, in `TransactionSerializer`.

Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/SignerForIdentifier.java:55]: Let me try to see if we can eliminate this method altogether.

Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/SignerForIdentifier.java:55]: @jframe I've eliminated this method. TransactionSerializer can reconstruct SignatureData from the hex String.

**Pull request merged**